utils/docs/gmake-caps-sym.1.in
.TH gmake-caps-sym 1 "@DATE@" "gEDA Project" @VERSION@

.SH NAME
gmake-caps-sym \- Make gEDA Capacitor Symbols

.SH SYNOPSIS
.B gmake-caps-sym
.RI [ OPTIONS ] TEMPLATE[.sym] PARAMETERS[.spm]

.SH DESCRIPTION
.B gmake-caps-sym
Make gEDA Capacitor Symbols

gmake-caps-sym is for 'batch' production of capacitor symbols for the geda-gaf project. The script requires a template symbol file and a parameter file. Parameters read from the file are used in the output symbol file name or used as replacement values for pre-set variables in the symbol template file.

.SH QUICK
.B usage: [-h, --help] | [--version] | [options] template[.sym] parameters[.spm]
example:
gmake-caps-sym electro-example-template.sym params-exported-gnumeric.spm<ENTER>.
.SH PARAMETERS
Fields in the TAB or space delimited parameter file are read and interpreted as follows:

1. Capacitence string to be substituted for 'CAPACITOR_VALUE' in the symbol file and the output file name
2. Voltage string is included in the file name and as part of the 'CAPACITOR_VALUE'
3. PartNumber string, included in the file name and in the 'PART_NUMBER'.
4. FootPrint string to be substituted for 'unknown' in the symbol file

.SH INFORMATION
Use of the following options displays information related to gmake-caps-sym
but does not create any symbols:
.IP "-h, --help"
Show information about the advanced usage of this command.
.IP "-u, --usage"
Displays examples on usage"
.IP "--version"
Displays software version information"

.SH DIRECTIVES
.IP "-v, --verbose"
Include additional information during processing.

.SH OPTIONS
The current version of gmake-caps-sym does not recognize any additional command-line options

.SH HOMEPAGE
http://www.geda-project.org

.SH BUGS
You can report bugs at the Launchpad site of gmake-caps-sym:
https://bugs.launchpad.net/geda/gmake-caps-sym/+filebug

.SH AUTHOR
Wiley Edward Hill < wileyhill dot gmail dot com >

.SH SEE ALSO
.BR gmake-transistors-sym (1) gmake-set-resistors (1)

.SH COPYRIGHT
.nf
Copyright \(co 2012-@YEAR@ Wiley Edward Hill.  License GPLv2+: GNU GPL
version 2 or later.  Please see the "COPYING" file included with this
program for full details.
.PP
This is free software: you are free to change and redistribute it. There is NO
WARRANTY; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE, to
the extent permitted by law.


NEWS
==============================================
 'gschem and Friends' Electronic Design Suite
==============================================

Copyright (C) 1998-2017 gEDA Developers

This file documents important user-visible changes in gEDA/gaf.  For
more information, please consult the `ChangeLog' file.

Notable changes in gEDA/gaf gEDA 2.1.1.1 - 04/01/19 Rampage Edition

* Expanded support for internationalization (i18n).

* All utilities now support internationalization (i18n).

* The visibility of toolbar tooltips can now be control using menu options.

* The context menu available when creating Polygon objects now includes
  an UNDO option allowing users to undo the previous section and resume
  path mode from the previous node.

* Scripting to support printing schematic documents from the command-line
  using gschem has been enhanced to support postscript or pdf output.
  Previously the default was changed from postscript to pdf. Postscript
  outputs could still be produced using menu options available in gschem.
  The output type is now automatically detected when using the command
  line interface by the file extension of the output file name. The
  default remains pdf if the output name is not given.

* Configuration function output-type has been renamed output-type.

* Libgedacairo now uses round end-caps when drawing buses, resulting in
  smoother transitions and improved appearance.

* A new keyword option hide-columns has been added to gattrib to allow
  user to designate a list attributes to be hidden in the GUI be default.
  A new Column Visibility dialog has been added to support user management
  of visible columns.

* A new path backend has been added to the gaf tool to allow scripting
  lanauages other than scheme to retrieve path information from libgeda.

* Improved appearence of various dialogs in gschem.

* gschem now logs exiting of the schematic editor if log-time is enabled.

* gschem now reports failure to attach attributes that aleady attached.

* gschem text properties dialog now supports operations to manipulate
  text case.

* gschem translate dialog now allows for translation of only selected
  objects. Additionally, the translate offset entry now accepts X and
  Y coordinates.

* Auxiliary buffers can now be explicitly cleared using a right context
  menu on the corresponding paste menu item.

* The suite now include template documents install to the user's template
  folder so the user can create new schematic and symbol documents from
  the system's context menu.

* Issues resolved:

* Libgedacairo has been enhanced to resolve issues with the end points of
  arcs shifting at different zoom levels. The shifting sometimes resulted
  in misalignment to adjoining lines at higher magnification levels.

* Extensive revisions were made to the Rampage Edition to build the suite
  on MinGW platforms. The Rampage Edition is now fully functional under
  native Win32 host systems including i18n support.

* The Rename Session dialog now fills in the previous session name,
  previously to dialog presented the user with an empty name field.

* The locked bit is not preserved when copying components in gschem.

* gschem now restarts text search when users change text in the Find Text
  dialog and click "find" or hits ENTER instead of closing the dialog.

* gschem now raises the main window when interacting with the Array Dialog.
  Previously, user could selected other application windows and return to
  the dialog and select the button that lowered the dialog but did not
  ensure the main window was in the foreground.

* An issue was resolved with controls not being initially enabled on the
  Attributes tab of the Preferences Dialog if the Attributes tab was the
  "saved" initial tab in gschem. Switching tabs or the view list controls
  re-enabled the disabled controls. The saved tab is now restored after
  settings are uploaded to the Preferences Dialog and this is reflected
  in the initial state of controls on the Attributes tab.

* An issue was resolved with resetting component-dialog-attributes when
  a second list was encountered in rc files. If a second attributes list
  was processed gschem would fault when the list was accessed or exiting
  gschem.

* The drc2 back end no longer reports duplicate slots for on components
  that do not have slots. This generally occurred when components have
  duplicate reference designators because these would typically show up
  using the same slot number.

* Libgeda no longer reports an oddity regarding symbol versions if the
  oddity was due to the symbol file being missing.

* The scheme file geda-deprecated-config.scm is no longer installed.

Notable changes in gEDA/gaf gEDA 2.1.1.1 - 11/12/17 Rampage Edition

* Expanded support for internationalization (i18n).

* The build system now support using the --with-included-gettext option to
  configure the suite to build and use an in-source international library.

* Font-sizes for widgets on GedaToolbars can now be specified in RC files.

* The toolbar layout in gschem has been improved to better support labels.

* The visibility of toolbar tooltips can now be control using RC keywords.

* The Slot Editor Dialog now incorporates a spinner control to adjust the
  slot number of slotted components. The range is restricted to valid slots
  for the selected component.

* New bitmaps have been added;

      * geda-deslect.png
      * geda-find-attribute
      * geda-properties.png
      * geda-trash-can.png

* libgedathon no longer over-writes existing gafrc when the declare_local_sym
  feature is used.

* libgedathon now supports internationalization (i18n).

Notable changes in gEDA/gaf gEDA 2.1.1.0 - 01/01/17 Rampage Edition

* Improvements were made to both the build system and source code to ensure
  gEDA-GAF builds more reliably on 64 bit platforms. Testing was performed
  using Ubuntu 12.04 and Ubuntu 14.04 64 bit host systems.


utils/gmake/gmake-caps-sym.sh
#!/bin/bash

VER=1.0

# Author: Wiley Edward Hill
# Date: 02/25/13
#
# Abstract:

MINARGS=1          # Script requires at least one arguments.
ERR_BAD_ARGS=65
VERBOSE=false
PrefixLength=3
Suxfix=".sym"

TemplateFile=""
ParameterFile=""
Prefix=4

vecho()
{
   if [[ ! $1 = "" ]] ; then
     if $VERBOSE ; then
      echo $1
     fi
   fi
}

do_show_usage(){
   echo Usage:   `basename $0` '[-h, --help] | [--version] | [options] template[.sym] parameters[.spm]'
}

do_process(){
   echo "Using Parameter File: $ParameterFile"
   while read -u3 line
   do
      parameters=( $line )
      Capacitance=${parameters[0]}
      Voltage=${parameters[1]}
      PartNumber=${parameters[2]}
      FootPrint=${parameters[3]}
      vecho "Read Capacitance=$Capacitance, Voltage=$Voltage, PartNumber=$PartNumber, FootPrint=$FootPrint"
      if [ ! "${Capacitance:0:1}" = "#" ] && [ ! "$Capacitance" = "" ] ; then
         OutputFile=$Prefix"_"$Capacitance"_"$Voltage"_"$PartNumber$Suxfix
         Value=$Capacitance", "$Voltage
         cat $TemplateFile | sed -e s/"PART_NUMBER"/"$PartNumber"/g >$OutputFile.tp1
         cat $OutputFile.tp1 | sed -e s/"unknown"/"$FootPrint"/g >$OutputFile.tp2
         cat $OutputFile.tp2 | sed -e s/"CAPACITOR_VALUE"/"$Value"/g >$OutputFile
         # ------ Remove the tmp files
         if [ -f $OutputFile.tp1 ]; then
           rm $OutputFile.tp1
         fi
         if [ -f $OutputFile.tp2 ]; then
           rm $OutputFile.tp2
         fi
      fi
   done 3< <(cat $ParameterFile)
}

if [ "$#" -lt $MINARGS ] ; then do_show_usage; exit $ERR_BAD_ARGS ; fi

if [ "$1" = "--version" ] ; then echo `basename $0` "Version $VER" ; exit 0 ; fi
if [ "$1" = "-h" ] || [ "$1" = "--help" ] ; then do_show_usage ; exit 0 ; fi
if [ "$1" = "-q" ] || [ "$1" = "--quiet" ] ; then quiet=true ; shift ; fi
if [ "$1" = "-v" ] || [ "$1" = "--verbose" ] ; then VERBOSE=true ; shift ; fi

if [ -f $1 ] ; then
  TemplateFile="$1"
  shift ;
else
  if [ -f "$1.sym" ] ; then
    TemplateFile="$1.sym"
  else
    echo "Template File not found: $TemplateFile"
    exit $ERR_BAD_ARGS
  fi
fi

Prefix=`echo "${TemplateFile}" | cut -d'-' -f1` ; # trim away all but first word

if [ -f $1 ] ; then
  ParameterFile= "$1"
  shift ;
else
  if [  -f "$1.spm" ] ; then
    ParameterFile="$1.spm"
  else
    echo "Parameter File not found: $ParameterFile"
    exit $ERR_BAD_ARGS
  fi
fi

do_process


















